### PR TITLE
Update SoundCloud-Feeds with Pagination

### PR DESCRIPTION
**Pull request by mprangenberg**
This uses Pagination as per the SoundCloud API: https://developers.soundcloud.com/docs/api/guide#pagination to get all episodes of a feed. See Issue #180.
I don't know if this needs to respect the max_episodes_per_feed-setting. If so, could someone point me to how i can get the value of that setting?

### CODE REVIEW

File: src/gpodder/plugins/soundcloud.py [hunk 1]
```diff
@@ -133,33 +133,37 @@ def get_tracks(self, feed):
...
-                    'link': track.get('permalink_url') or 'http://soundcloud.com/'+self.username,
-                    'description': track.get('description') or _('No description available'),
-                    'url': url,
-                    'file_size': int(filesize),
-                    'mime_type': filetype,
-                    'guid': track.get('permalink', track.get('id')),
-                    'published': soundcloud_parsedate(track.get('created_at', None)),
-                }
+
+            while not json_url == '':
```
Comment (thp) [inline src/gpodder/plugins/soundcloud.py]: `while json_url != '':`

**Comment by thp** [comment]
Holding off merging this for now, since it increases the API requests. We might want to let the user provide their soundcloud API key like we do for YouTube so we don't hit API request limits.

**Comment by vossad01** [comment]
Depends on #179

**Comment by elelay** [comment]
Soundcloud doesn't let one register applications anymore.
So I'm rejecting this PR as it could increase API requests.

**Pull request closed without merging**